cosign.py

```diff
@@ -118,57 +149,116 @@ def sign_image(
     payload_size = len(payload)
     payload_digest = f'sha256:{hashlib.sha256(payload).hexdigest()}'
 
-    oci_client.put_blob(
-        image_reference=signature_image_reference,
-        digest=payload_digest,
-        octets_count=payload_size,
-        data=payload,
-    )
+    if on_exist is OnExist.APPEND and exists:
+        manifest = oci_client.manifest(
+            image_reference=signature_image_reference,
+        )
+        for layer_idx, layer in enumerate(manifest.layers):
+            a = layer.annotations
+            existing_signature = a.get('dev.cosignproject.cosign/signature', None)
```

Comment:
I'd prefer to factor-out this annotation as well (just like the public key and signature annotation). Or you could also introduce a new enum like `SignatureAnnotations` or something like that.